src/citrine/resources/file_link.py

```diff
@@ -357,6 +372,81 @@ def get(self,
 
         raise NotFound(f"Found file, but no version {version}")
 
+    def get2(self,
+             uid: Union[UUID, str],
+             *,
+             version: Optional[Union[UUID, str, int]] = None) -> FileLink:
+        """
+        Get an element of the collection by its id.
+
+        Parameters
+        ----------
+        uid: Union[UUID, str]
+            A representation of the FileLink (Citrine id or file name)
+        version: Optional[UUID, str, int]
+            The version, as a UUID or str(UUID) of the version_id or an int or
+            str(int) of the version number.  If None, returns the file with the
+            highest version number (most recent).
+
+        Returns
+        -------
+        ResourceType
+            An object with specified scope and uid
+
+        """
+        if not isinstance(uid, (str, UUID)):
+            raise TypeError(f"File Link can only be resolved from str or UUID."
+                            f"Instead got {type(uid)} {uid}.")
+        if version is not None and not isinstance(version, (str, UUID, int)):
+            raise TypeError(f"Version can only be resolved from str, int or UUID."
+                            f"Instead got {type(uid)} {uid}.")
+
+        try:  # Check if the uid string is actually a UUID
+            if isinstance(uid, str):
+                uid = UUID(uid)
+        except ValueError:
+            pass
+
+        try:  # Check if the version string is actually a UUID
+            if isinstance(version, str):
+                version = UUID(version)
+        except ValueError:
+            pass
+
+        try:  # Check if the version string is actually an int / version number
+            if isinstance(version, str):
+                version = int(version)
+        except ValueError:
+            pass
+
+        if isinstance(version, str):
+            raise ValueError(
+                f"Version {version} could not be converted to either an int or a UUID"
+            )
```

Comment:
These checks are not independent:
```suggestion
        if isinstance(version, str):
            try:  # Check if the version string is actually a UUID
                version = UUID(version)
            except ValueError:
                try:  # Check if the version string is actually an int / version number
                    version = int(version)
                except ValueError:
                    raise ValueError(
                        f"Version {version} could not be converted to either an int or a UUID"
                    )
```